judge.pl

```diff
@@ -1225,17 +1244,23 @@ sub main_loop {
                 $judge->finish_job($r->{job_id}, $cats::job_st_failed);
             }
             else {
+                $r->{split_strategy} = get_split_strategy($r);
                 my $problem = $judge->get_problem($r->{problem_id});
+
                 if ($problem->{run_method} == $cats::rm_competitive ||
-                    $r->{type} == $cats::job_type_submission_part || !$judge->can_split ||
-                    ($r->{req_job_split_strategy} // $r->{cp_job_split_strategy} // '') eq 'none') {
-                        test_problem($r, $problem)
-                    }
-                    else {
-                        my $state = split_solution($r);
-                        $state ? $judge->set_request_state($r, $state, $current_job_id) :
-                            test_problem($r, $problem);
-                    }
+                    $r->{type} == $cats::job_type_submission_part ||
+                    $r->{split_strategy}->{method} eq $cats::split_none) {
+                        test_problem($r, $problem);
+                }
+                elsif (!$judge->can_split) {
+                    log_msg("Can't split solution: queue limit reached\n");
```

Comment:
No, this is not the only reason. For example, local judge just returns false since splitting does not make sense for it.
You can return reason from can_split or just log from inside it (logger is passed to judge backend).